src/MensaWidget/MWFrame.class.st

```diff
@@ -11,43 +11,160 @@ Class {
+
+	^ 450 @ 550
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:03'
+}
+MWFrame class >> contentMorphColor [
+
+	^ Color darkGray
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 00:54'
+}
+MWFrame class >> contentMorphInnerBorder [
+
+	^ 5
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:50'
+}
+MWFrame class >> contentPosition [
+
+	^ 5 @ 40
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Louis Christ 6/6/2024 19:50'
+}
+MWFrame class >> frameSize [
+
+	^ 460 @ 595
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 01:01'
+}
+MWFrame class >> scrollBarThickness [
+
+	^ 20
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 00:57'
+}
+MWFrame class >> spacingBetweenCanteenMorphs [
+
+	^ 5
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 01:13'
+}
+MWFrame class >> windowHeaderSize [
+
+	^ 40 
+]
+
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 5/29/2024 18:43'
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:37'
 }
-MWFrame >> addText [
+MWFrame >> buildFrameContent [
 
-	| textMorph |
+	| contentMorph |
 	
-	textMorph := TextMorph new.
-	textMorph contents: 'Are you a hungry?';
-		position:  135@100.
-	self addPaneMorph: textMorph.
+	contentMorph := ScrollPane new.
+	self fillContentMorph: contentMorph.
+	contentMorph scrollBarThickness: self class scrollBarThickness;
+		color: self class contentMorphColor;
+		position: self class contentPosition;
+		extent: self class contentExtent.
+	self addPaneMorph: contentMorph
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 5/27/2024 17:00'
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:42'
 }
-MWFrame >> createFrame [
-		
-	self setLabel:'Mensa Widget';
-		openInWorld;
-		extent: 400@600.	
+MWFrame >> fillContentMorph: aContentMorph [
 	
+	| contentMorph |
 	
-	
-
+	contentMorph := AlignmentMorph newColumn.
+	contentMorph cellInset: self class cellSpacing;
+		hResizing: #shrinkWrap;
+		vResizing: #shrinkWrap.
+	MWDataGettingWrapper getTodaysMeals do: [ :each | 
+		contentMorph addMorph: each getCanteenMorph].
+	aContentMorph scroller addMorph: contentMorph
 	
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 5/27/2024 17:00'
+	#'squeak_changestamp' : 'tmc 6/6/2024 22:45'
 }
 MWFrame >> initialize [
 	
 	super initialize.
-	self createFrame;
-		addText.
+	self setLabel:'Mensa Widget';
+		buildFrameContent;
+		openInWorld;
+		extent: self class frameSize;
+		removeMaximize;
+		removeScalability
+		 "Cascade expected ->"
+]
+
+{
+	#category : #creation,
+	#'squeak_changestamp' : 'tmc 6/6/2024 22:53'
+}
+MWFrame >> removeMaximize [
+	
+	self submorphsDo: [ :each | 
+	each class = AlignmentMorph ifTrue: [
+		each submorphsDo: [ :each2 |
+			( each2 class = SystemWindowButton and:
+				[ each2 actionSelector = #expandBoxHit ]) 
+					ifTrue: [ each2 abandon ]]]]
```

Comment:
do we do spaces after "(" too? and why suddenly not do the space before "]"? I think adding a space before "]" would make it very ugly, but if we don't do them, we're inconsistent. Why not abandon the space thing, no normal text uses spaces after brackets and smalltalk shall be readable like a normal text, let's talk about it in the meeting.